Add Java interop section (#3502)
SonarSource/rspec: rules/S6518/kotlin/rule.adoc

because the semantics of indexed access operators is evident to the reader,
while for a function call, the reader would need to know what the called function does.

==== Java interop

> The issue is also raised when Java API is used as a Kotlin compiler treats all `get(.)` methods as possible index access operators. So it's tecnically possible to write `CompletableFuture[timeout, unit] instead of `CompletableFuture.get(timeout, unit)` when using `CompletableFuture` from Java.

Reviewer: Spelling mistake in "tecnically" and missing backtick. Alternative suggestion:
```suggestion
The issue is also raised when Java API is used as a Kotlin compiler treats all `get(.)` methods as possible index access operators. Use `CompletableFuture[timeout, unit]` instead of `CompletableFuture.get(timeout, unit)` when accessing Java's `CompletableFuture`.
```
